worlds/apquest/test/test_hammer.py

```diff
@@ -0,0 +1,59 @@
+from typing import cast
+
+from test.param import classvar_matrix
+
+from ..items import APQuestItem
+from .bases import APQuestTestBase
+
+# This file assumes that you've read test_extra_starting_chest.py first.
+# classvar_matrix will not be explained again here, so make sure you've read that first.
+
+# We'll once again use classvar_matrix to test that the world behaves correctly both if "hammer" option is disabled,
+# and also that it behaves correctly when the "hammer" option is enabled.
+hammer_options = [
+    {"hammer": False},
+    {"hammer": True},
+]
+
+
+@classvar_matrix(options=hammer_options)
+class TestHammer(APQuestTestBase):
+    # The hammer option adds the Hammer item to the itempool, so let's verify that it's there (only when it should be).
+    # Remember: Each test here will get run twice, once with the "hammer" option disabled, and once with it enabled.
+    # We need to check which case we're in, and react appropriately.
+    def test_hammer_exists(self) -> None:
+        if self.world.options.hammer:
+            # To test whether an item exists, we need to find it in the itempool
+            # Note: The Item class overrides __eq__, so you could do this using TestCase.assertIn with an Item instance
+            # However, the author dislikes this pattern and prefers checking the necessary conditions inside a next().
+            with self.subTest("Test that Hammer item exists and is progression if hammer option is enabled"):
+                hammer = next((item for item in self.multiworld.itempool if item.name == "Hammer"), None)
+                self.assertIsNotNone(hammer)
+                self.assertTrue(cast(APQuestItem, hammer).advancement)
```

Comment:
what does this cast do? it's not APQuestItem that defines the `advancement` field, right?